### ISSUE TITLE
Reset files every time

### ISSUE THREAD

**Issue by bryx** [SamKirkland/FTP-Deploy-Action#124]
**Bug Description**
I encounter a reset every time I make a commit and thus a bad synchronization between files on GitHub and in my FTP folder on an OVH server. I'm developing a Symfony application, there are many files on the /var/ and /vendor/ of Symfony that are managed as bash commands in my workflow.

Example :
- Creation of folders then files
- Putting the files online

Another example :
- creation of index.php => it is present in FTP
- deletion of index.php => it is not deleted by FTP

How to have a good sync with your tool that for the blow is really easy to understand. Is there a parameter that I'm missing although I've already looked at tickets on this repository.

**My Action Config**
```yaml
name: 🚀 OVHcloud

on:
  push:
    branches: [ master ]

jobs:
  web-deploy:
    name: 🎉 Envoi sur le serveur
    runs-on: ubuntu-latest

    steps:
    - name: 🚚 Get latest code
      uses: actions/checkout@v2.3.2
      with:
        lfs: true
        fetch-depth : 100 

    - name: 📂 Sync files
      uses: SamKirkland/FTP-Deploy-Action@4.0.0
      with:
        server: ftp.cluster023.hosting.ovh.net
        username: aaa
        password: bbb
        server-dir: test/
        exclude: .git* 
          - .git*/** 
          - .vscode/**
          - .github/**

    - name: ls -a via ssh
      uses: garygrossgarten/github-action-ssh@release
      with:
        host: ssh.cluster023.hosting.ovh.net
        username: aaa
        password: bbb
        command: |
          cd test/
          curl -sS https://getcomposer.org/installer | /usr/local/php7.3/bin/php -d allow_url_fopen=On
          /usr/local/php7.3/bin/php -d allow_url_fopen=on -d memory_limit=-1 composer.phar install
```

**My Action Log**
```
It is never created in the test folder on FTP
On the other hand visible on the action of GitHub

Run SamKirkland/FTP-Deploy-Action@4.0.0
----------------------------------------------------------------
🚀 Thanks for using ftp-deploy version 1.0.0. Let's deploy some stuff!   
----------------------------------------------------------------
If you found this project helpful, please support it
by giving it a ⭐ on Github --> https://github.com/SamKirkland/FTP-Deploy-Action
or add a badge 🏷️ to your projects readme --> https://github.com/SamKirkland/FTP-Deploy-Action#badge
----------------------------------------------------------------
No file exists on the server "test/.ftp-deploy-sync-state.json" - this much be your first publish! 🎉
The first publish will take a while... but once the initial sync is done only differences are published!
If you get this message and its NOT your first publish, something is wrong.
----------------------------------------------------------------
Local Files:	166
Server Files:	166
----------------------------------------------------------------
Calculating differences between client & server
----------------------------------------------------------------
➕ Upload: .env
➕ Upload: .htaccess
➕ Upload: bclover.png
📁 Create: bin
➕ Upload: bin/console
➕ Upload: composer.json
➕ Upload: composer.lock
➕ Upload: composer.phar
📁 Create: config
➕ Upload: config/bundles.php
📁 Create: config/packages
➕ Upload: config/packages/cache.yaml
📁 Create: config/packages/dev
➕ Upload: config/packages/dev/debug.yaml
➕ Upload: config/packages/dev/monolog.yaml
➕ Upload: config/packages/dev/web_profiler.yaml
➕ Upload: config/packages/doctrine_migrations.yaml
➕ Upload: config/packages/doctrine.yaml
➕ Upload: config/packages/framework.yaml
➕ Upload: config/packages/knp_paginator.yaml
➕ Upload: config/packages/mailer.yaml
➕ Upload: config/packages/notifier.yaml
📁 Create: config/packages/prod
➕ Upload: config/packages/prod/deprecations.yaml
➕ Upload: config/packages/prod/doctrine.yaml
➕ Upload: config/packages/prod/monolog.yaml
➕ Upload: config/packages/prod/routing.yaml
➕ Upload: config/packages/routing.yaml
➕ Upload: config/packages/sensio_framework_extra.yaml
📁 Create: config/packages/test
➕ Upload: config/packages/test/framework.yaml
➕ Upload: config/packages/test/monolog.yaml
➕ Upload: config/packages/test/twig.yaml
➕ Upload: config/packages/test/validator.yaml
➕ Upload: config/packages/test/web_profiler.yaml
➕ Upload: config/packages/translation.yaml
➕ Upload: config/packages/twig.yaml
➕ Upload: config/packages/validator.yaml
➕ Upload: config/preload.php
📁 Create: config/routes
➕ Upload: config/routes.yaml
➕ Upload: config/routes/annotations.yaml
📁 Create: config/routes/dev
➕ Upload: config/routes/dev/framework.yaml
➕ Upload: config/routes/dev/web_profiler.yaml
➕ Upload: config/services.yaml
➕ Upload: index.php
📁 Create: public
➕ Upload: public/.htaccess
➕ Upload: public/index.php
➕ Upload: README.md
📁 Create: src
➕ Upload: src/Bclover.php
📁 Create: src/Controller
➕ Upload: src/Controller/MembreController.php
➕ Upload: src/Controller/PortfolioController.php
➕ Upload: src/Controller/PrincipalController.php
➕ Upload: src/Controller/UniversController.php
📁 Create: src/Entity
📁 Create: src/Entity/Blog
➕ Upload: src/Entity/Blog/Article.php
➕ Upload: src/Entity/Blog/Categorie.php
➕ Upload: src/Entity/Blog/Commentaire.php
📁 Create: src/Entity/Forums
➕ Upload: src/Entity/Forums/Forum.php
➕ Upload: src/Entity/Forums/Reponse.php
➕ Upload: src/Entity/Forums/Sujet.php
📁 Create: src/Entity/Membres
➕ Upload: src/Entity/Membres/Avertir.php
➕ Upload: src/Entity/Membres/Bannir.php
➕ Upload: src/Entity/Membres/Compte.php
➕ Upload: src/Entity/Membres/Droit.php
➕ Upload: src/Entity/Membres/Jeton.php
➕ Upload: src/Entity/Membres/Membre.php
📁 Create: src/Entity/Personnages
📁 Create: src/Entity/Personnages/Anniversaire
➕ Upload: src/Entity/Personnages/Anniversaire/Jour.php
➕ Upload: src/Entity/Personnages/Anniversaire/Mois.php
➕ Upload: src/Entity/Personnages/Compagnie.php
➕ Upload: src/Entity/Personnages/Personnage.php
➕ Upload: src/Entity/Personnages/Royaume.php
➕ Upload: src/Entity/Personnages/Specialite.php
➕ Upload: src/Entity/Web.php
📁 Create: src/Form
➕ Upload: src/Form/ContactFormType.php
📁 Create: src/Form/Membres
➕ Upload: src/Form/Membres/ConnexionFormType.php
➕ Upload: src/Form/Membres/InscriptionFormType.php
➕ Upload: src/Form/Membres/PasswordFormType.php
➕ Upload: src/Form/Membres/ProfilFormType.php
➕ Upload: src/Form/Membres/ProfilFormTypeAvatar.php
➕ Upload: src/Form/Membres/ProfilFormTypeAvertir.php
➕ Upload: src/Form/Membres/ProfilFormTypeBannir.php
➕ Upload: src/Form/Membres/ProfilFormTypeEdit.php
➕ Upload: src/Form/Membres/ProfilFormTypeEmail.php
➕ Upload: src/Form/Membres/ProfilFormTypePassword.php
📁 Create: src/Form/Personnages
➕ Upload: src/Form/Personnages/PersonnageFormType.php
➕ Upload: src/Kernel.php
📁 Create: src/Migrations
➕ Upload: src/Migrations/bclover.sql
📁 Create: src/Repository
📁 Create: src/Repository/Blog
➕ Upload: src/Repository/Blog/ArticleRepository.php
➕ Upload: src/Repository/Blog/CategorieRepository.php
➕ Upload: src/Repository/Blog/CommentaireRepository.php
📁 Create: src/Repository/Forums
➕ Upload: src/Repository/Forums/ForumRepository.php
➕ Upload: src/Repository/Forums/ReponseRepository.php
➕ Upload: src/Repository/Forums/SujetRepository.php
📁 Create: src/Repository/Membres
➕ Upload: src/Repository/Membres/AvertirRepository.php
➕ Upload: src/Repository/Membres/BannirRepository.php
➕ Upload: src/Repository/Membres/CompteRepository.php
➕ Upload: src/Repository/Membres/DroitRepository.php
➕ Upload: src/Repository/Membres/JetonRepository.php
➕ Upload: src/Repository/Membres/MembreRepository.php
📁 Create: src/Repository/Personnages
📁 Create: src/Repository/Personnages/Anniversaire
➕ Upload: src/Repository/Personnages/Anniversaire/JourRepository.php
➕ Upload: src/Repository/Personnages/Anniversaire/MoisRepository.php
➕ Upload: src/Repository/Personnages/CompagnieRepository.php
➕ Upload: src/Repository/Personnages/PersonnageRepository.php
➕ Upload: src/Repository/Personnages/RoyaumeRepository.php
➕ Upload: src/Repository/Personnages/SpecialiteRepository.php
➕ Upload: src/Repository/WebRepository.php
📁 Create: src/Templates
➕ Upload: src/Templates/base.html.twig
📁 Create: src/Templates/emails
➕ Upload: src/Templates/emails/contact.html.twig
📁 Create: src/Templates/emails/membre
➕ Upload: src/Templates/emails/membre/password.html.twig
📁 Create: src/Templates/pages
➕ Upload: src/Templates/pages/contact.html.twig
➕ Upload: src/Templates/pages/index.html.twig
📁 Create: src/Templates/pages/membre
➕ Upload: src/Templates/pages/membre/compte.html.twig
➕ Upload: src/Templates/pages/membre/connexion.html.twig
➕ Upload: src/Templates/pages/membre/index.html.twig
➕ Upload: src/Templates/pages/membre/inscription.html.twig
➕ Upload: src/Templates/pages/membre/password.html.twig
➕ Upload: src/Templates/pages/membre/profil.html.twig
➕ Upload: src/Templates/pages/offline.html.twig
➕ Upload: src/Templates/pages/page.html.twig
📁 Create: src/Templates/pages/portfolio
📁 Create: src/Templates/pages/portfolio/autres
➕ Upload: src/Templates/pages/portfolio/autres/benevolats.html.twig
➕ Upload: src/Templates/pages/portfolio/autres/benevolats.modals.html.twig
➕ Upload: src/Templates/pages/portfolio/autres/stages.html.twig
➕ Upload: src/Templates/pages/portfolio/autres/stages.modals.html.twig
➕ Upload: src/Templates/pages/portfolio/index.html.twig
📁 Create: src/Templates/pages/portfolio/projets
➕ Upload: src/Templates/pages/portfolio/projets/index.html.twig
➕ Upload: src/Templates/pages/portfolio/projets/modals.html.twig
📁 Create: src/Templates/pages/portfolio/veilles
➕ Upload: src/Templates/pages/portfolio/veilles/index.html.twig
➕ Upload: src/Templates/pages/portfolio/veilles/modals.html.twig
📁 Create: src/Templates/pages/univers
➕ Upload: src/Templates/pages/univers/index.html.twig
➕ Upload: src/Templates/pages/univers/personnage.html.twig
➕ Upload: src/Templates/pages/univers/personnage.view.html.twig
➕ Upload: src/Templates/widget.html.twig
📁 Create: src/Translates
➕ Upload: src/Translates/KnpPaginatorBundle.en.yaml
➕ Upload: symfony.lock
➕ Upload: www.json
----------------------------------------------------------------
Making changes to 166 files to sync server state
Uploading: 3.09 MB -- Deleting: 0 B -- Replacing: 0 B
----------------------------------------------------------------
creating folder "bin/"
creating folder "config/"
creating folder "config/packages/"
creating folder "config/packages/dev/"
creating folder "config/packages/prod/"
creating folder "config/packages/test/"
creating folder "config/routes/"
creating folder "config/routes/dev/"
creating folder "public/"
creating folder "src/"
creating folder "src/Controller/"
creating folder "src/Entity/"
creating folder "src/Entity/Blog/"
creating folder "src/Entity/Forums/"
creating folder "src/Entity/Membres/"
creating folder "src/Entity/Personnages/"
creating folder "src/Entity/Personnages/Anniversaire/"
creating folder "src/Form/"
creating folder "src/Form/Membres/"
creating folder "src/Form/Personnages/"
creating folder "src/Migrations/"
creating folder "src/Repository/"
creating folder "src/Repository/Blog/"
creating folder "src/Repository/Forums/"
creating folder "src/Repository/Membres/"
creating folder "src/Repository/Personnages/"
creating folder "src/Repository/Personnages/Anniversaire/"
creating folder "src/Templates/"
creating folder "src/Templates/emails/"
creating folder "src/Templates/emails/membre/"
creating folder "src/Templates/pages/"
creating folder "src/Templates/pages/membre/"
creating folder "src/Templates/pages/portfolio/"
creating folder "src/Templates/pages/portfolio/autres/"
creating folder "src/Templates/pages/portfolio/projets/"
creating folder "src/Templates/pages/portfolio/veilles/"
creating folder "src/Templates/pages/univers/"
creating folder "src/Translates/"
uploading ".env"
uploading ".htaccess"
uploading "bclover.png"
uploading "bin/console"
uploading "composer.json"
uploading "composer.lock"
uploading "composer.phar"
uploading "config/bundles.php"
uploading "config/packages/cache.yaml"
uploading "config/packages/dev/debug.yaml"
uploading "config/packages/dev/monolog.yaml"
uploading "config/packages/dev/web_profiler.yaml"
uploading "config/packages/doctrine_migrations.yaml"
uploading "config/packages/doctrine.yaml"
uploading "config/packages/framework.yaml"
uploading "config/packages/knp_paginator.yaml"
uploading "config/packages/mailer.yaml"
uploading "config/packages/notifier.yaml"
uploading "config/packages/prod/deprecations.yaml"
uploading "config/packages/prod/doctrine.yaml"
uploading "config/packages/prod/monolog.yaml"
uploading "config/packages/prod/routing.yaml"
uploading "config/packages/routing.yaml"
uploading "config/packages/sensio_framework_extra.yaml"
uploading "config/packages/test/framework.yaml"
uploading "config/packages/test/monolog.yaml"
uploading "config/packages/test/twig.yaml"
uploading "config/packages/test/validator.yaml"
uploading "config/packages/test/web_profiler.yaml"
uploading "config/packages/translation.yaml"
uploading "config/packages/twig.yaml"
uploading "config/packages/validator.yaml"
uploading "config/preload.php"
uploading "config/routes.yaml"
uploading "config/routes/annotations.yaml"
uploading "config/routes/dev/framework.yaml"
uploading "config/routes/dev/web_profiler.yaml"
uploading "config/services.yaml"
uploading "index.php"
uploading "public/.htaccess"
uploading "public/index.php"
uploading "README.md"
uploading "src/Bclover.php"
uploading "src/Controller/MembreController.php"
uploading "src/Controller/PortfolioController.php"
uploading "src/Controller/PrincipalController.php"
uploading "src/Controller/UniversController.php"
uploading "src/Entity/Blog/Article.php"
uploading "src/Entity/Blog/Categorie.php"
uploading "src/Entity/Blog/Commentaire.php"
uploading "src/Entity/Forums/Forum.php"
uploading "src/Entity/Forums/Reponse.php"
uploading "src/Entity/Forums/Sujet.php"
uploading "src/Entity/Membres/Avertir.php"
uploading "src/Entity/Membres/Bannir.php"
uploading "src/Entity/Membres/Compte.php"
uploading "src/Entity/Membres/Droit.php"
uploading "src/Entity/Membres/Jeton.php"
uploading "src/Entity/Membres/Membre.php"
uploading "src/Entity/Personnages/Anniversaire/Jour.php"
uploading "src/Entity/Personnages/Anniversaire/Mois.php"
uploading "src/Entity/Personnages/Compagnie.php"
uploading "src/Entity/Personnages/Personnage.php"
uploading "src/Entity/Personnages/Royaume.php"
uploading "src/Entity/Personnages/Specialite.php"
uploading "src/Entity/Web.php"
uploading "src/Form/ContactFormType.php"
uploading "src/Form/Membres/ConnexionFormType.php"
uploading "src/Form/Membres/InscriptionFormType.php"
uploading "src/Form/Membres/PasswordFormType.php"
uploading "src/Form/Membres/ProfilFormType.php"
uploading "src/Form/Membres/ProfilFormTypeAvatar.php"
uploading "src/Form/Membres/ProfilFormTypeAvertir.php"
uploading "src/Form/Membres/ProfilFormTypeBannir.php"
uploading "src/Form/Membres/ProfilFormTypeEdit.php"
uploading "src/Form/Membres/ProfilFormTypeEmail.php"
uploading "src/Form/Membres/ProfilFormTypePassword.php"
uploading "src/Form/Personnages/PersonnageFormType.php"
uploading "src/Kernel.php"
uploading "src/Migrations/bclover.sql"
uploading "src/Repository/Blog/ArticleRepository.php"
uploading "src/Repository/Blog/CategorieRepository.php"
uploading "src/Repository/Blog/CommentaireRepository.php"
uploading "src/Repository/Forums/ForumRepository.php"
uploading "src/Repository/Forums/ReponseRepository.php"
uploading "src/Repository/Forums/SujetRepository.php"
uploading "src/Repository/Membres/AvertirRepository.php"
uploading "src/Repository/Membres/BannirRepository.php"
uploading "src/Repository/Membres/CompteRepository.php"
uploading "src/Repository/Membres/DroitRepository.php"
uploading "src/Repository/Membres/JetonRepository.php"
uploading "src/Repository/Membres/MembreRepository.php"
uploading "src/Repository/Personnages/Anniversaire/JourRepository.php"
uploading "src/Repository/Personnages/Anniversaire/MoisRepository.php"
uploading "src/Repository/Personnages/CompagnieRepository.php"
uploading "src/Repository/Personnages/PersonnageRepository.php"
uploading "src/Repository/Personnages/RoyaumeRepository.php"
uploading "src/Repository/Personnages/SpecialiteRepository.php"
uploading "src/Repository/WebRepository.php"
uploading "src/Templates/base.html.twig"
uploading "src/Templates/emails/contact.html.twig"
uploading "src/Templates/emails/membre/password.html.twig"
uploading "src/Templates/pages/contact.html.twig"
uploading "src/Templates/pages/index.html.twig"
uploading "src/Templates/pages/membre/compte.html.twig"
uploading "src/Templates/pages/membre/connexion.html.twig"
uploading "src/Templates/pages/membre/index.html.twig"
uploading "src/Templates/pages/membre/inscription.html.twig"
uploading "src/Templates/pages/membre/password.html.twig"
uploading "src/Templates/pages/membre/profil.html.twig"
uploading "src/Templates/pages/offline.html.twig"
uploading "src/Templates/pages/page.html.twig"
uploading "src/Templates/pages/portfolio/autres/benevolats.html.twig"
uploading "src/Templates/pages/portfolio/autres/benevolats.modals.html.twig"
uploading "src/Templates/pages/portfolio/autres/stages.html.twig"
uploading "src/Templates/pages/portfolio/autres/stages.modals.html.twig"
uploading "src/Templates/pages/portfolio/index.html.twig"
uploading "src/Templates/pages/portfolio/projets/index.html.twig"
uploading "src/Templates/pages/portfolio/projets/modals.html.twig"
uploading "src/Templates/pages/portfolio/veilles/index.html.twig"
uploading "src/Templates/pages/portfolio/veilles/modals.html.twig"
uploading "src/Templates/pages/univers/index.html.twig"
uploading "src/Templates/pages/univers/personnage.html.twig"
uploading "src/Templates/pages/univers/personnage.view.html.twig"
uploading "src/Templates/widget.html.twig"
uploading "src/Translates/KnpPaginatorBundle.en.yaml"
uploading "symfony.lock"
uploading "www.json"
----------------------------------------------------------------
🎉 Sync complete. Saving current server state to "test/.ftp-deploy-sync-state.json"
Error 553 FileNameNotAllowed, you don't have access to upload that file
FTPError: 553 Can't open that file: No such file or directory
    at FTPContext._onControlSocketData (/home/runner/work/_actions/SamKirkland/FTP-Deploy-Action/4.0.0/dist/index.js:3760:39)
    at Socket.<anonymous> (/home/runner/work/_actions/SamKirkland/FTP-Deploy-Action/4.0.0/dist/index.js:3601:44)
    at Socket.emit (events.js:210:5)
    at addChunk (_stream_readable.js:309:12)
    at readableAddChunk (_stream_readable.js:286:13)
    at Socket.Readable.push (_stream_readable.js:224:10)
    at TCP.onStreamRead (internal/stream_base_commons.js:182:23) {
  name: 'FTPError',
  code: 553
}
FTPError: 553 Can't open that file: No such file or directory
    at FTPContext._onControlSocketData (/home/runner/work/_actions/SamKirkland/FTP-Deploy-Action/4.0.0/dist/index.js:3760:39)
    at Socket.<anonymous> (/home/runner/work/_actions/SamKirkland/FTP-Deploy-Action/4.0.0/dist/index.js:3601:44)
    at Socket.emit (events.js:210:5)
    at addChunk (_stream_readable.js:309:12)
    at readableAddChunk (_stream_readable.js:286:13)
    at Socket.Readable.push (_stream_readable.js:224:10)
    at TCP.onStreamRead (internal/stream_base_commons.js:182:23) {
  name: 'FTPError',
  code: 553
}
----------------------------------------------------------------
Time spent hashing:               94 milliseconds
Time spent connecting to server:  894 milliseconds
Time spent deploying:             1 minute 14.2 seconds (41.7 kB/second)
  - changing dirs:                18.8 seconds
  - logging:                      98 milliseconds
----------------------------------------------------------------
Total time:                       1 minute 15.7 seconds
----------------------------------------------------------------
```

Thanks ;)
PS : I'm a French user, my English is so bad, sorry!

**Comment by SamKirkland** [comment]
Hello!

It looks like the main issue might be #120 - which I'm working on a fix for.

It appears you are running composer on your server, I'm not familiar with composer but you might want to run that before you trigger a sync. See this --> https://github.com/marketplace/actions/composer-php-actions

Also you can delete `fetch-depth : 100` version 4 no longer needs it!

I'll look into getting a fix out today and will update this post so you can try it out.

**Comment by SamKirkland** [comment]
I've published a patch for version `4.0.0` that should resolve this issue.
If you re-run your action you should see the issue go away 😃

**Comment by bryx** [comment]
Thank you, your patch is working on my side!
I am very happy to see that you were very responsive.
Thanks again for your support.
